testing/test_metal_context.mm

```diff
@@ -17,43 +17,47 @@
 namespace flutter {
 
 TestMetalContext::TestMetalContext() {
-  auto device = fml::scoped_nsprotocol{MTLCreateSystemDefaultDevice()};
+  id<MTLDevice> device = MTLCreateSystemDefaultDevice();
   if (!device) {
     FML_LOG(ERROR) << "Could not acquire Metal device.";
     return;
   }
 
-  auto command_queue = fml::scoped_nsobject{[device.get() newCommandQueue]};
+  id<MTLCommandQueue> command_queue = [device newCommandQueue];
   if (!command_queue) {
     FML_LOG(ERROR) << "Could not create the default command queue.";
     return;
   }
 
-  [command_queue.get() setLabel:@"Flutter Test Queue"];
+  [command_queue setLabel:@"Flutter Test Queue"];
 
   GrMtlBackendContext backendContext = {};
   // Skia expect arguments to `MakeMetal` transfer ownership of the reference in for release later
   // when the GrDirectContext is collected.
-  backendContext.fDevice.reset([device.get() retain]);
-  backendContext.fQueue.reset([command_queue.get() retain]);
+  backendContext.fDevice.reset((__bridge_retained GrMTLHandle)device);
+  backendContext.fQueue.reset((__bridge_retained GrMTLHandle)command_queue);
   skia_context_ = GrDirectContexts::MakeMetal(backendContext);
   if (!skia_context_) {
     FML_LOG(ERROR) << "Could not create the GrDirectContext from the Metal Device "
                       "and command queue.";
   }
 
-  device_ = [device.get() retain];
-  command_queue_ = [command_queue.get() retain];
+  // Retain and transfer to non-ARC-managed pointers.
+  device_ = (__bridge_retained void*)device;
+  command_queue_ = (__bridge_retained void*)command_queue;
 }
 
 TestMetalContext::~TestMetalContext() {
   std::scoped_lock lock(textures_mutex_);
   textures_.clear();
   if (device_) {
-    [(__bridge id)device_ release];
+    // Release and transfer to ARC-managed pointer.
+    id<MTLDevice> device = (__bridge_transfer id<MTLDevice>)device_;
+    device = nil;
```

Comment:
This line is necessary to stop the compiler yelling about unused objects, but maybe it kind of drives home the point that hey, now no one has a reference to the underlying object anymore. 🤷‍♂️

Reply:
Worth adding a comment of what you said here.